### ISSUE TITLE
`newline-per-chained-call` and PropTypes

### ISSUE THREAD

**Issue by Spy-Seth** [airbnb/javascript#748]
Hi everyone,

I'm facing a some trouble with one rule: how do you define `PropTypes` with the `newline-per-chained-call` rule activated?

```javascript
/* @flow */

import React from 'react';

class Baz extends React.Component {
    static propTypes = {
        foo: React.PropTypes.oneOf(['nice', 'bad']),  // This trigger an error
        bar: React.PropTypes.instanceOf(Date).isRequired, // Here too
    };

    // ...
}
```

**Comment by lencioni** [comment]
If I understand [the rule](http://eslint.org/docs/rules/newline-per-chained-call) correctly, and [the configuration we have here](https://github.com/airbnb/javascript/blob/b648de3233c157c81b9d6301d061e16136ca1d29/packages/eslint-config-airbnb/rules/style.js#L62-L65), I would expect your `foo` example to not trigger an error but your bar example to trigger an error, since we have set `ignoreChainWithDepth` to `3`.

In which case, you should be able to resolve this by first destructuring `PropTypes` from `React`, and then using it directly:

```js
/* @flow */

import React from 'react';

const { PropTypes } = React;

class Baz extends React.Component {
    static propTypes = {
        foo: PropTypes.oneOf(['nice', 'bad']),
        bar: PropTypes.instanceOf(Date).isRequired,
    };

    // ...
}
```

**Comment by Spy-Seth** [comment]
Ho you right! I've build this example from actual code, and only the `bar` prop types trigger error. My bad.

If the only solution is to play with the `PropTypes` object, I guess this is the trick:

```javascript 
import { default as React, PropTypes } from 'react';
```

**Comment by KyleHodgetts** [comment]
@Spy-Seth What does `default as React` achieve?
Does it just save you having to do:

`import React from 'react';`
`import { PropTypes } from React;`

?

**Comment by lencioni** [comment]
I believe that this syntax might happen work:

```js
import React, { PropTypes } from 'react';
```

However, since `PropTypes` is not (yet?) an actual named export (since React isn't using ES2015 modules), I think that this is not supposed work according to spec (even though it may happen to work with your environment). I think it is probably best to destructure separately for now, to avoid potential problems when upgrading your transpiler or module bundler.

I'd like to throw in a plug for [import-js](https://github.com/trotzig/import-js), an editor plugin I've been working on that helps you add imports to your files. Check it out! :)

**Comment by ljharb** [comment]
I'm starting to think this particular rule might be a bit odd - it's called "newline per chained call" but it's counting property access (which is not a "call"). imo none of those examples should be throwing. @ yannickcr, what do you think?

**Comment by dustinmoorenet** [comment]
There is an issue opened on eslint for that rule. It appears to be a bug. https://github.com/eslint/eslint/issues/5289

IMO, this rule should be turned off until that issue is resolved.

**Comment by ljharb** [comment]
Agreed.

**Comment by ljharb** [comment]
Published as v6.0.1.

We'll re-enable this rule once `eslint` has fixed the bug.

**Comment by Spy-Seth** [comment]
That's solution resolve my issue :) :+1:

**Comment by taion** [comment]
This got fixed in 2.3.0, BTW.

**Comment by taion** [comment]
https://github.com/eslint/eslint/commit/9b54ed3e767b9f718703ca030f02d6f96c1a4c84
